lib/SILOptimizer/Mandatory/TFCanonicalizeCFG.h

```diff
@@ -95,6 +96,24 @@ namespace tf {
     }
   };
 
+  /// Represents an SESERegion that is shared between different SESERegions.
+  class FunctionSESERegion : public SESERegionTree {
```

Comment:
this is not really a function. how about SharedRegion?

Reply:
I beg to differ. We are really handling a shared region (this is the WHAT of what we do). Whether it's implemented by lowering to a function, or duplicating the blocks (the HOW), is an impl detail, which is subject to change in the future.

It is especially confusing currently, as we are now lowering to a function...